reporter/collector_reporter.go

```diff
@@ -113,7 +113,13 @@ func (r *CollectorReporter) GetMetrics() Metrics {
 func (r *CollectorReporter) reportProfile(ctx context.Context) error {
 	traceEvents := r.traceEvents.WLock()
 	events := maps.Clone(*traceEvents)
+	originsMap := make(map[libpf.Origin]samples.KeyToEventMapping, 2)
 	clear(*traceEvents)
+	for _, origin := range []libpf.Origin{support.TraceOriginSampling,
+		support.TraceOriginOffCPU} {
+		originsMap[origin] = make(samples.KeyToEventMapping)
```

Comment:
Why not `clear(*traceEvents[origin])` here?

Reply:
Pushed https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/319/commits/5a06b724bb03471df5b2b02ce23ce732ad3c64a0 with this approach to reduce the number of allocations.